src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,49 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function desopo_pape_shortest_paths(g::AbstractGraph, 
+    src::Integer,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real
+    U = eltype(g)
+    nvg = nv(g)
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
+    state = Vector{UInt8}()
+    for i=1:nvg
+        push!(state, 2)
+    end
+    q = Vector{U}()
+    push!(q, src)
+    @inbounds dists[src] = 0
```

Comment:
We will get into trouble here, if `src` is not in the range `1:nvg`. You could for example add a test for that at the beginning of the function, and if it not the case, then throw a `DomainError`.

Reply:
Sure!

Reply:
I don't think we check this anywhere else:
```
julia> g = Graph(10,20)
{10, 20} undirected simple Int64 graph

julia> dijkstra_shortest_paths(g, 100)
ERROR: BoundsError: attempt to access 10-element Array{Int64,1} at index [100]
```